pmd-core/src/main/java/net/sourceforge/pmd/util/database/DBMSMetadata.java

```diff
@@ -371,29 +336,16 @@ public List<SourceObject> getSourceObjectList(List<String> languages, List<Strin
                         returnSourceCodeObjectsStatement);
                 try (PreparedStatement sourceCodeObjectsStatement = getConnection()
                         .prepareStatement(returnSourceCodeObjectsStatement)) {
-                    for (String language : searchLanguages) {
-                        for (String schema : searchSchemas) {
-                            for (String sourceCodeType : searchSourceCodeTypes) {
-                                for (String sourceCodeName : searchSourceCodeNames) {
-                                    sourceObjectsList.addAll(findSourceObjects(sourceCodeObjectsStatement, language, schema,
-                                            sourceCodeType, sourceCodeName));
-                                }
-                            }
-                        }
-                    }
+                    generateSourceObjects(sourceObjectsList, sourceCodeObjectsStatement, searchLanguages, searchSchemas,
+                            searchSourceCodeTypes, searchSourceCodeNames);
                 }
             } else {
                 // Use standard DatabaseMetaData interface
-                LOG.debug(
-                        "Have dbUri - no returnSourceCodeObjectsStatement, reverting to DatabaseMetaData.getProcedures(...)");
+                LOG.debug("Have dbUri - no returnSourceCodeObjectsStatement, reverting to DatabaseMetaData.getProcedures(...)");
```

Comment:
Putting a series of for loops into a separate method improves the readability and maintainability of the code. It is a very good idea. Acceptable!